packages/firebase_auth/firebase_auth/example/android/app/src/test/java/io/flutter/plugins/firebaseauth/FirebaseAuthPluginTest.java

```diff
@@ -0,0 +1,77 @@
+package io.flutter.plugins.firebaseauth;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
+
+import android.util.SparseArray;
+import androidx.test.core.app.ApplicationProvider;
+import io.flutter.embedding.engine.plugins.FlutterPlugin;
+import io.flutter.plugin.common.BinaryMessenger;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.PluginRegistry;
+import io.flutter.view.FlutterMain;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.robolectric.RobolectricTestRunner;
+
+@RunWith(RobolectricTestRunner.class)
+public class FirebaseAuthPluginTest {
+  @Mock PluginRegistry.Registrar mockRegistrar;
+  @Mock BinaryMessenger mockBinaryMessenger;
+  @Mock FlutterPlugin.FlutterPluginBinding mockFlutterPluginBinding;
+  @Mock MethodChannel.Result mockResult;
+
+  private String testFirebaseAppName = "[DEFAULT]";
+
+  FirebaseAuthPlugin plugin;
+
+  @BeforeClass()
+  public static void BeforeClass() {
+    FlutterMain.setIsRunningInRobolectricTest(true);
```

Comment:
I've seen you use this in the `google_maps` plugin but not in `image_picker`. I cannot figure out how to run tests without calling this and I know it is deprecated :(

Reply:
I don't recall entirely but I think this is required for certain robotic methods. If the test fails without this, then you probably need it.

Reply:
@cyanglaz Gotcha!